gadgets/cmr/CmrRealTimeLAXCineAIAnalysisGadget.cpp

```diff
@@ -15,6 +14,77 @@
 
 namespace Gadgetron { 
 
+    // a singleton class to load model
+    // the current finding is the python runtime may not always release memory after loading in the model and related libraries
+    // as a result, repeated model loading (e.g. from multiple calls of this chain) will increase VM usage
+    // although not fully resolving the problem, making the model loading a singleton can mitigate this problme
+
+    class CmrRealTimeLAXCineAIModel
+    {
+    public:
+
+        static void get_model_singleton(const std::string& model_home, const std::string& model_file, boost::python::object& model);
+      
+    protected:
+        CmrRealTimeLAXCineAIModel() = default;
+        static CmrRealTimeLAXCineAIModel* instance_;
+        static std::map<std::string, boost::python::object> models_;
```

Comment:
I think you need this to key on both home and file name, since you would otherwise be able to change home but not the file name in a call and get the wrong model.

Reply:
fixed